src/app/my-rewards/tx-history/components/Table/TransactionHistoryHeaderRow.tsx

```diff
@@ -110,6 +116,40 @@ export const HeaderCell = ({
 }
 
 export const TransactionHistoryHeaderRow = (): ReactElement => {
+  const { rows } = useTableContext<ColumnId, TransactionHistoryCellDataMap>()
+
+  const visibleTotalAmountUsd = useMemo(() => {
+    if (!rows.length) return ''
+
+    let sum = 0
+
+    const addValue = (raw: string) => {
+      if (!raw) return
+      const sanitized = raw.replace(/,/g, '').replace(/^</, '')
+      const parsed = Number(sanitized)
+      if (!Number.isNaN(parsed)) {
+        sum += parsed
+      }
+    }
+
+    rows.forEach(row => {
```

Comment:
Instead of manually looping and updating sum, consider using `reduce`.

```
const sum = rows.reduce((acc, row) => {
  const usd = row.data.total_amount.usd
  const values = Array.isArray(usd) ? usd : [usd]
  return acc + values.reduce((subtotal, val) => subtotal + toNumber(val), 0)
}, 0)

const toNumber = (raw: string | null | undefined) => {
  if (!raw) return 0
  const sanitized = raw.replace(/,/g, '').replace(/^</, '')
  const parsed = Number(sanitized)
  return Number.isNaN(parsed) ? 0 : parsed
}
```

Btw the `toNumber` function can be easily moved to `utils.ts`.